### DISCUSSION TITLE
DB2 setup fails when installing license

### DISCUSSION THREAD

**Discussion by rboccadoro** [HCL-TECH-SOFTWARE/connections-automation#47] [Q&A] [answered]
When I run the setup I get this
fatal: [conn7.eldeng.it]: FAILED! => {"changed": true, "cmd": ["/home/db2inst1/sqllib/adm/db2licm", "-a", "/opt/IBM/InstallBinaries/ese_u/db2/license/db2ese_u.lic"], "delta": "0:00:00.038202", "end": "2021-01-18 17:04:14.813610", "msg": "non-zero return code", "rc": 127, "start": "2021-01-18 17:04:14.775408", "stderr": "/home/db2inst1/sqllib/adm/db2licm: error while loading shared libraries: libdb2locale.so.1: cannot open shared object file: No such file or directory", "stderr_lines": ["/home/db2inst1/sqllib/adm/db2licm: error while loading shared libraries: libdb2locale.so.1: cannot open shared object file: No such file or directory"], "stdout": "", "stdout_lines": []}

**Comment by rboccadoro** [comment]
I checked and the file exist. I am running the playbook as root.

**Reply by dejo1307** [reply]
What version of OS/distribution is that? I never saw it before but can try to reproduce it

**Comment by rboccadoro** [comment]
CentOS Linux release 7.8.2003 (Core)

**Reply by dejo1307** [reply]
Just to confirm that I understand it correctly - you are not using basically db2inst1 as the owner for DB2, but you are using root user instead?

**Reply by dejo1307** [reply]
And DB2 installation passed, but license part only is failing? Asking only since I am refactoring this role for those cases.

**Comment by rboccadoro** [comment]
I am running this as root: ansible-playbook -i /root/connections-automation-main/environments/examples/cnx7/connections_7_without_component_pack/connections playbooks/setup-connections-complete.yml 

The installation is successful, only the license part is failing

**Reply by dejo1307** [reply]
ok, that shouldn't make the difference. looking into it

**Reply by dejo1307** [reply]
Do you mind sending me the output of "localectl status" from that machine?

**Comment by rboccadoro** [comment]
localectl status
   System Locale: LANG=en_US.UTF-8
       VC Keymap: it
      X11 Layout: it

**Comment by rboccadoro** [comment, marked as answer]
Any news Dejan ?

**Reply by dejo1307** [reply]
Hi Roberto,

Actually, I tried on a couple of different flavors, and didn't manage to reproduce that. So to make it easier, I prepared actually this: https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/main/documentation/howtos/local_development_with_vagrant.md (but forgot to update it here). 

So if you don't mind, let's try two things:

- First, on your machine, just run yum update, and run ansible. See if it fails on licence (I am using the files from Flexnet myself).
- If you don't mind, just try the stuff with Vagrant from the link I posted above. That would basically spin up the local VM and install in first two steps DB2 and the licence, and just see if it works. That is at least something that is 100% 1:1 on whoever's local machine. 

This has something to do with a machine itself, but I have no other ideas how to troubleshoot this. We can of course, if it doesn't work, try to jump on a call, maybe 5 minutes of call can move us somewhere if nothing else works.

**Comment by rboccadoro** [comment]
Hi Dejan

The first suggestion worked. Did a yum update and then DB2 installed 
correctly including the license.
I have still an error

TASK [ldap-install : Install OpenLDAP] 
*****************************************
included: 
/root/connections-automation-main/roles/third_party/ldap-install/tasks/install_openldap.yml 
for conn7.eldeng.it

TASK [ldap-install : Install OpenLDAP and dependencies on CentOS] 
**************
skipping: [conn7.eldeng.it]

TASK [ldap-install : Install OpenLDAP and dependencies on RHEL] 
****************
ok: [conn7.eldeng.it]

TASK [ldap-install : Start slapd service] 
**************************************
fatal: [conn7.eldeng.it]: FAILED! => {"changed": false, "msg": "failure 1 
during daemon-reload: Failed to execute operation: Connection timed 
out\n"}

PLAY RECAP 
*********************************************************************
conn7.eldeng.it            : ok=66   changed=24   unreachable=0 failed=1  
skipped=13   rescued=0    ignored=1 

I will keep you updated, in the meantime thank you very much for your help

**Reply by dejo1307** [reply]
Ok, good point. I will document it a bit better, because I have no idea why, but I managed to see in at least 50% of out of the box CentOS (and RHEL) 7 boxes totally random errors when it comes to DB2 installation which were fixed after yum update.

Regarding other issue, can you just run playbooks/third_party/setup-ldap.yml playbook against the same inventory file? It timed out for some reason (just SSH connection), just hit this to go through. After it would come playbooks/third_party/setup-tdi.yml.
